adafruit_stmpe610.py

```diff
@@ -250,54 +373,149 @@ def _read_register(self, register, length):
+     Defaults to None.
+    :param int disp_rotation: display rotation in degrees. Values allowed are
+     0, 90, 180, and 270. Defaults to 0.
+    :param (bool, bool) touch_flip: swap touchscreen axis range minimum and
+     maximum values for (x, y) axes as referenced to display 0-degree rotation.
+     Defaults to (False, False).
+
+    ** Quickstart: Importing and instantiating Adafruit_STMPE610_I2C**
+
+    Import the Adafruit_STMPE610_SPI class and instantiate for the 2.4" TFT Wing
+    after instantiating the display:
+
+    .. code-block:: python
+
+        import adafruit_stmpe610
+        ts = adafruit_stmpe610.Adafruit_STMPE610_SPI(spi, cs=cs_pin,
+            baudrate=1000000,
+            calibration=((357, 3812), (390, 3555)),
+            size=(display.width, display.height), disp_rotation=display.rotation,
+            touch_flip=(False, False))
+
     """
 
-    def __init__(self, spi, cs, baudrate=1000000):
-        """
-        Check the STMPE610 was found,Default clock rate 1000000 - can be changed with 'baudrate'
-        """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        spi,
+        cs,
+        baudrate=1000000,
+        calibration=None,
+        size=None,
+        disp_rotation=0,
+        touch_flip=(False, False),
+    ):
+
+        self._calib = calibration
+        self._disp_size = size
+        self._disp_rotation = disp_rotation
+        self._touch_flip = touch_flip
+
+        # Check the touchscreen calibration and display size kwargs.
+        if not self._calib:
+            self._calib = ((0, 4095), (0, 4095))
+        if not self._disp_size:
+            self._disp_size = (4095, 4095)
+
+        if self._disp_rotation not in (0, 90, 180, 270):
+            raise ValueError("Display rotation value must be 0, 90, 180, or 270")
+
+        # Check that the STMPE610 was found.
         import adafruit_bus_device.spi_device as spidev  # pylint: disable=import-outside-toplevel
 
         self._spi = spidev.SPIDevice(spi, cs, baudrate=baudrate)
         # Check device version.
         version = self.get_version
         if _STMPE_VERSION != version:
-            # if it fails try SPI MODE 1  -- that is what Arduino does
+            # If it fails try SPI MODE 1  -- that is what Arduino does
             self._spi = spidev.SPIDevice(
                 spi, cs, baudrate=baudrate, polarity=0, phase=1
             )
             version = self.get_version
             if _STMPE_VERSION != version:
                 raise RuntimeError(
-                    "Failed to find STMPE610! Chip Version 0x%x. "
+                    "Failed to find STMPE610 controller! Chip Version 0x%x. "
                     "If you are using the breakout, verify you are in SPI mode."
                     % version
                 )
         super().__init__()
 
+    @property
+    def touch_point(self):  # pylint: disable=too-many-branches
```

Comment:
See above: Why does touch_point need to be replicated for I2C and SPI?

Reply:
The first submitted version placed touch_point only in the parent class, but pylint didn't like it. It didn't recognize  screen_size, etc. as members of the parent class. At least the touch_point duplicated version will be easy to change if the I2C class is removed.